src/lightning_app/core/app.py

```diff
@@ -307,9 +307,11 @@ def maybe_apply_changes(self) -> bool:
         if not deltas:
             # When no deltas are received from the Rest API or work queues,
             # we need to check if the flow modified the state and populate changes.
-            if Delta(DeepDiff(self.last_state, self.state, verbose_level=2)).to_dict():
+            deep_diff = DeepDiff(self.last_state, self.state, verbose_level=2)
```

Comment:
Database updates won't trigger deltas?

Reply:
You mean in the future when we add support for databases?

Reply:
Yes

Reply:
Good question :-) Possibly yes but it depends on the shape things take:
- if we have a database as a work component, then I would say it's up to the work to trigger an update by modifying a state variable when a transaction happens (or when it makes sense)
- if we have a database-backed state, then things definitely get more interesting, because the database is itself (part of) the state, so it should trigger a rerun automatically